bats/fb-test-katello.bats

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bats
+# vim: ft=sh:sw=2:et
+
+set -o pipefail
+
+load os_helper
+load foreman_helper
+
+setup() {
+  tSetOSVersion
+}
+
+@test "check web app is up" {
+  curl -sk "https://localhost$URL_PREFIX/users/login" | grep -q login-form
+}
+
+@test "wake up puppet agent" {
+  source ~/.bashrc
+  puppet agent -t -v
+}
+
+@test "check web app is still up" {
+  curl -sk "https://localhost$URL_PREFIX/users/login" | grep -q login-form
+}
+
+@test "check smart proxy is registered" {
+  count=$(hammer --csv proxy list | wc -l)
+  [ $count -gt 1 ]
```

Comment:
I wonder if grepping for the hostname would be a better check, but I think this is directly ported so it's fine to leave this in.